src/client/src/shell/components/app-header/Header.tsx

```diff
@@ -5,16 +5,73 @@ import { styled, ThemeName, useTheme } from 'rt-theme'
 import Logo from './Logo'
 
 class Header extends React.Component {
+  // The added buttons handle the Spot windows management
+
+  // The state handles the buttons visibility
+  state = {
+    showStackAll: false,
+    showTabAll: false,
+    showToggleCollapse: false,
+  }
+
+  componentDidMount() {
+    // On Spot window open/close the buttons visibility is updated
+    const interval = setInterval(() => {
```

Comment:
Polling like this in the middle of a component to probe if glue42's API availability is not ideal, especially inside a 
UI component.

Is the global glue42 variable not available immediately?  If it is, Glue42 should have some method that exposes a promise indicating the `glue.agm.register` is available. You should then use an async method in `componentDidMount` to set the state

Reply:
Also maybe worth using hooks instead of the old component lifecycle methods